src/main/webapp/script.js

```diff
@@ -29,7 +32,14 @@ function getGeolocation() {
 function displayLocation(position) {
   let lat = position.coords.latitude;
   let lng = position.coords.longitude;
-  fetchList('/data?lat=' + lat + "&lng=" + lng);
+  locationQuery = '/data?lat=' + lat + '&lng=' + lng;
+  product = document.getElementById('product').value;
+  if (product === '') {
+    fetchList(locationQuery + '&product=NULL');
+  }
```

Comment:
Would it be simpler to leave this empty. and use .equals("") in SmallCityService?

Reply:
yes it is. I just changed it.